File: ch01/cpp/triangle.cpp
#include <cstdio>
#include <algorithm>

const int MAX_N = 100;
int n, edge[MAX_N];

int solve() {
  int max_perimeter = 0;

  // 棒は重複して選ばないようにする
  for (int i = 0; i < n; i++) {
    for (int j = i + 1; j < n; j++) {
      for (int k = j + 1; k < n; k++) {
        int perimeter = edge[i] + edge[j] + edge[k];  // 周囲の長さ
        int longest_edge = std::max(edge[i], std::max(edge[j], edge[k])); // 最長の辺
        int rest = perimeter - longest_edge;  // その他2辺の和
        
        if (rest > longest_edge) {
          max_perimeter = std::max(max_perimeter, perimeter);
        }
      }
    }
  }

  return max_perimeter;
}

int main() {
  scanf("%d", &n);
  for (int i = 0; i < n; i++) {
    scanf("%d", &edge[i]);
  }

  int max_perimeter = solve();

  printf("%d\n", max_perimeter);

  return 0;
}

File: ch01/python/triangle.py
MAX_N = 100

# 標準入力から n, edge を取得
n = int(input())
edge = []
for i in range(n):
    edge.append(int(input()))

# 最大周長の初期化
max_perimeter = 0

for i in range(0, n):
    for j in range(i + 1, n):
        for k in range(j + 1, n):
            perimeter = edge[i] + edge[j] + edge[k]
            longest = max(edge)
            rest = perimeter - longest

            if rest > longest:
                max_perimeter = max([max_perimeter, perimeter])

print(max_perimeter)
